aviatrix/resource_aviatrix_dcf_mwp_policy_list.go

```diff
@@ -0,0 +1,458 @@
+	var smartGroups []string
+
+	smartGroupsSet, ok := policyMap[key].(*schema.Set)
+	if !ok {
+		return nil, fmt.Errorf("%s must be of type *schema.Set", key)
+	}
+
+	for _, smartGroup := range smartGroupsSet.List() {
+		smartGroupStr, ok := smartGroup.(string)
+		if !ok {
+			return nil, fmt.Errorf("%s must be of type string", key)
+		}
+
+		smartGroups = append(smartGroups, smartGroupStr)
+	}
+
+	return smartGroups, nil
+}
+
+func resourceAviatrixDCFPolicyListCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, ok := meta.(*goaviatrix.Client)
+	if !ok {
+		return diag.Errorf("client must be of type *goaviatrix.Client")
+	}
+
+	policyList, err := marshalDCFPolicyListInput(d)
+	if err != nil {
+		return diag.Errorf("invalid inputs for Distributed-firewalling Policy during create: %s\n", err)
+	}
+
+	uuid, err := client.CreateDCFPolicyList(ctx, policyList)
+	if err != nil {
+		return diag.Errorf("failed to create Distributed-firewalling Policy List: %s", err)
+	}
+
+	d.SetId(uuid)
+
+	return nil
+}
+
+//nolint:funlen,cyclop
+func resourceAviatrixDCFPolicyListRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, ok := meta.(*goaviatrix.Client)
+	if !ok {
+		return diag.Errorf("client must be of type *goaviatrix.Client")
+	}
+
+	uuid := d.Id()
+
+	policyList, err := client.GetDCFPolicyList(ctx, uuid)
+	if err != nil {
+		if errors.Is(err, goaviatrix.ErrNotFound) {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("failed to read Distributed-firewalling Policy List: %s", err)
+	}
+
+	var policies []map[string]interface{}
+	for _, policy := range policyList.Policies {
+		if policy.SystemResource {
+			continue
+		}
+		p := make(map[string]interface{})
+		p["name"] = policy.Name
+		p["action"] = policy.Action
+		p["priority"] = policy.Priority
+		p["src_smart_groups"] = policy.SrcSmartGroups
+		p["dst_smart_groups"] = policy.DstSmartGroups
+		p["web_groups"] = policy.WebGroups
+		p["logging"] = policy.Logging
+		p["watch"] = policy.Watch
+		p["uuid"] = policy.UUID
+		p["exclude_sg_orchestration"] = policy.ExcludeSgOrchestration
+
+		if strings.EqualFold(policy.Protocol, "PROTOCOL_UNSPECIFIED") {
+			p["protocol"] = "ANY"
+		} else {
+			p["protocol"] = policy.Protocol
+		}
+		p["flow_app_requirement"] = policy.FlowAppRequirement
+		p["decrypt_policy"] = policy.DecryptPolicy
+
+		if policy.Protocol != "ICMP" {
+			var portRanges []map[string]interface{}
+			for _, portRange := range policy.PortRanges {
+				portRangeMap := map[string]interface{}{
+					"hi": portRange.Hi,
+					"lo": portRange.Lo,
+				}
+				portRanges = append(portRanges, portRangeMap)
+			}
+			p["port_ranges"] = portRanges
+		}
+		p["tls_profile"] = policy.TLSProfile
+
+		policies = append(policies, p)
+	}
+
+	if err := d.Set("name", policyList.Name); err != nil {
+		return diag.Errorf("failed to set name during Distributed-firewalling Policy List read: %s\n", err)
```

Comment:
here an elesewhere you don't need these newline characters in the print.  Terraform handles this.